nuxeo-distribution/nuxeo-launcher/src/main/java/org/nuxeo/launcher/NuxeoLauncher.java

```diff
@@ -1262,6 +1277,96 @@ public void run() {
         }
     }
 
+    protected boolean registerRemoteInstance() throws IOException, ConfigurationException, PackageException {
+        String username;
+        String password;
+        String project;
+        String description = "";
+        NuxeoClientInstanceType type;
+        if (System.console() != null) {
+            username = LauncherRegisterPromptHelper.promptUsername();
+            password = LauncherRegisterPromptHelper.promptPassword();
+            List<ConnectProject> projs = getConnectRegistrationBroker().getAvailableProjects(username, password);
+            project = LauncherRegisterPromptHelper.promptProjectId(projs);
+            type = LauncherRegisterPromptHelper.promptInstanceType();
+            description = LauncherRegisterPromptHelper.promptDescription();
+        } else {
+            String[] input = new String(IOUtils.toByteArray(System.in)).split("\n");
+            if (input.length < 4) {
+                throw new IOException("Wrong number of arguments.");
+            }
+            username = input[0];
+            password = input[1];
+            project = input[2];
+            type = NuxeoClientInstanceType.fromString(input[3]);
+            if (input.length > 3) {
+                description = StringUtils.join(Arrays.copyOfRange(input, 4, input.length), "\n");
+            }
+        }
+
+        if (type == null) {
+            type = NuxeoClientInstanceType.DEV;
+        }
```

Comment:
I'd agree with you; but as the value is not used anywhere except stored... I prefer to ask you to `create a ticket` and we'll handling it when it will be relevant and we know what we really want to do with the InstanceType.